### ISSUE TITLE
1st sample from tf.multinomial is always zero in webgpu backend

### ISSUE THREAD

**Issue by alvinleung1996** [tensorflow/tfjs#8057]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow.js): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): macOS 13.6.1 (22G313)
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow.js installed from (npm or script link): script link
- TensorFlow.js version (use command below): 4.12.0
- Browser version: Version 119.0.6045.105 (Official Build) (x86_64)
- Tensorflow.js Converter Version: N/A

**Describe the current behavior**

When sampling from `tf.multinomial` using the webgpu backend, the 1st sample is always zero.

**Describe the expected behavior**

All samples should respect the the given logits.

**Standalone code to reproduce the issue**

Execute the following code in the console in https://js.tensorflow.org/api/4.12.0/

```js
await tf.setBackend('webgpu')

let sum = tf.zeros([10])

const logits = tf.tensor1d([0.1, 0.9]).log()
for (let i = 0; i < 1000; i++) {
  const samples = tf.multinomial(logits, 10)
  sum = sum.add(samples)
}

sum.print()
```

**Other info / logs** 

webgpu backend:

```
Tensor
    [0, 894, 886, 900, 900, 912, 887, 877, 896, 880]
```

webgl backend:

```
Tensor
    [895, 901, 896, 884, 914, 894, 908, 906, 907, 912]
```

**Comment by gaikwadrahul8** [comment]
Hi, @alvinleung1996 

Thank you for bringing this issue to our attention and I tried to replicate the same issue from my end and I'm also getting same behavoiur which you mentioned in the issue template so we'll have to dig more into this issue and will update you soon. Thank you!

For reference I have added screenshot below:

`WebGPU Backend output :`

![image](https://github.com/tensorflow/tfjs/assets/115997457/47cb5236-ce34-4761-90ab-884d1dd61e00)

`WebGL Backend output :`

![image](https://github.com/tensorflow/tfjs/assets/115997457/346f59f3-f075-4d45-9a0a-57d7ed396066)

**Comment by adriankucharski** [comment]
This bug still exists in version 4.22.0 of the `@tensorflow/tfjs-backend-webgpu.`

**Comment by nileshau-afk** [comment]
Hello @alvinleung1996 ,

Thank you for bringing this to our attention and for providing a comprehensive overview. We appreciate the time you invested in creating this issue.

Currently, our internal development team is concentrated on other scheduled milestones, so we cannot prioritize this task immediately. However, we would love to see this move forward as a community-led effort.

If you’d like to submit a fix, please fork the repo and open a PR following the [TensorFlow Contribution Guidelines](https://www.tensorflow.org/community/contribute), [Tfjs Contributions ](https://github.com/tensorflow/tfjs/blob/master/tfjs-vis/CONTRIBUTING.md#how-to-contribute)and be sure to link this issue in your PR to avoid duplicate work, kindly leave a comment here if you plan to take this on.

Thank you again for your interest and time.

**Comment by Varun-S10** [comment]
Hi @alvinleung1996, Thank you for your patience. We were able to reproduce the issue. As a workaround, you can request one additional sample from the distribution and discard the first element using a standard tensor slice operation. Could you please test this out and let us know if it resolves the problem for your use case?
